interface/ceed-basis.c

```diff
@@ -1231,28 +1231,30 @@ int CeedBasisDestroy(CeedBasis *basis) {
 
   @ref Utility
 **/
-int CeedHaleTrefethenStripMap(CeedInt Q, CeedInt rho, CeedScalar *g, CeedScalar *g_prime) {
-  CeedScalar tau, d, C, PI2, PI = 4.0*atan(1.0);
+int CeedHaleTrefethenStripMap(CeedInt Q, CeedScalar rho, CeedScalar *g, CeedScalar *g_prime) {
+  CeedScalar i, tau, d, C, PI2, PI = 4.0*atan(1.0);
+  CeedScalar u[Q];
   tau = PI / log(rho);
   d = .5 + 1. / (exp(tau * PI) + 1.0)
   PI2 = PI / 2.0;
+  // u = asin(s)
   for (int i = 0; i < Q; i++) {
-    u = (PI * i) / Q;
+    u[i] = (asin(PI) * i) / Q;
     // Unscaled function of u
-    g_u = log(1.0 + exp(-tau * (PI2 + u))) - log(1.0 + exp(-tau * (PI2 - u))) + d * tau * u;
-    g_prime_u = 1.0 / (exp(tau * (PI2 + u)) + 1) + 1.0 / (exp(tau * (PI2 - u)) + 1) - d;
-    // Normalizing factor and scaled function
-    C = 1.0 / g_u;
-    g = C * g_u;
-    g_prime = -tau * C ;
+    g_u[i] = log(1.0 + exp(-tau * (PI2 + u[i]))) - log(1.0 + exp(-tau * (PI2 - u[i]))) + d * tau * u[i];
+    g_prime_u[i] = 1.0 / (exp(tau * (PI2 + u[i])) + 1) + 1.0 / (exp(tau * (PI2 - u[i])) + 1) - d;
+    // Normalizing factor and scaled function of s
+    C = 1.0 / log(1.0 + exp(-tau * (PI2 + PI2))) - log(1.0 + exp(-tau * (PI2 - PI2))) + d * tau * PI2;
+    g[i] = C * g_u[i];
+    g_prime[i] = -tau * C / sqrt(1.0 - sin(u[i]) * sin(u[i])) * g_prime_u[u[i]];
   }
   return CEED_ERROR_SUCCESS;
 }
 
 /**
   @brief Transform quadrature by applying a smooth mapping = (g, g_prime)
 
-  @param Q  Number of quadarture points
+  @param Q  Number of quadrature points
```

Comment:
In general, every argument has to be documented. I don't know what you intend this function to do; maybe it can be deleted or maybe it needs to take a `CeedBasis` as input and create a `CeedBasis` as output.